automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/methodconfigs/WorkspaceMethodConfigDetailsPage.scala

```diff
@@ -175,7 +197,7 @@ class WorkspaceMethodConfigDetailsPage(namespace: String, name: String, methodCo
   // TODO This is a very weak check to deterimine if page is ready
   def isLoaded: Boolean = {
     await spinner "Loading attributes..."
-    openLaunchAnalysisModalButton.isVisible
+    openLaunchAnalysisModalButton.isStateDisabled || openLaunchAnalysisModalButton.isStateEnabled
```

Comment:
`isStateDisabled` and `isStateEnabled` checks values of two attributes: `isVisible` and `data-test-state`.

Reply:
if they're testing two separate things, one of them should be renamed; right now this reads as if the value of one should always be the inverted value of the other, which makes this line extremely confusing as i think it'll always evaluate to true

Reply:
or at the very least, put this comment in the actual code so people can see the explanation outside this PR

Reply:
yes, I know what you mean, it is confusing... I'll try to clarify further. Using `isStateDisabled` or `isStateEnabled` forces WebDriver to wait for the button has rendered attribute `data-test-state` then look for attribute value. I don't care what the value of `data-test-state` is, only cares whether it is found in button.

Reply:
not always evaluate to true because if `isVisible` is false, it will evaluate to false.

Reply:
+1 to @helgridly's comments. If this is really testing a condition of "the button is rendered on the page and has a value in its `data-test-state`  attribute", I suggest making a new method on the `Button`  component that encapsulates such logic.

Reply:
okay, but at least put such explanations in the code, not this PR. i won't be the only person confused by this... the explanation should live with the code. and i still think that changing the names of the functions to better reflect what they do is the best way to deal with this :)

Reply:
revised method to remove confusion.